src/app/zap-templates/templates/app/attribute-size-src.zapt

```diff
@@ -0,0 +1,139 @@
+{{> header}}
+
+#include <app/util/af.h>
+#include <app/util/basic-types.h>
+#include <support/SafeInt.h>
+#include <support/logging/CHIPLogging.h>
+
+#include "gen/af-structs.h"
+
+using namespace chip;
+
+// The first 2 bytes specify the number of entries. A value of 0xFFFF means the list in invalid
+// and data is undefined.
+constexpr uint16_t kSizeLengthInBytes = 2u;
+
+void copyListMember(uint8_t * dest, uint8_t * src, bool write, uint16_t * offset, uint16_t length)
+{
+    if (write)
+    {
+        memmove(dest + *offset, src, length);
+    }
+    else
+    {
+        memmove(dest, src + *offset, length);
+    }
+
+    *offset = static_cast<uint16_t>(*offset + length);
+}
+
+uint16_t emberAfCopyList(ClusterId clusterId, EmberAfAttributeMetadata * am, bool write, uint8_t * dest, uint8_t * src, int32_t index)
+{
+    if (index == -1)
+    {
+        memmove(dest, src, am->size);
+        return am->size;
+    }
+
+    if (index == 0)
+    {
+       uint16_t entryCount = emberAfGetInt16u(src, 0, kSizeLengthInBytes);
+       emberAfCopyInt16u(dest, 0, entryCount);
+       return kSizeLengthInBytes;
+    }
+
+    if (!chip::CanCastTo<uint16_t>(index))
+    {
+        ChipLogError(Zcl, "Index %l is invalid. Should be between 1 and 65534", index);
+        return 0;
+    }
+
+    uint16_t entryLength = 0;
+    switch (clusterId)
+    {
+ {{#chip_server_clusters}}
+ {{#if (chip_has_list_attributes)}}
+    case {{asHex code 4}}: // {{name}} Cluster
+    {
+        uint16_t entryOffset = kSizeLengthInBytes;
+        switch (am->attributeId)
+        {
+        {{#chip_server_cluster_attributes}}
+        {{#if isList}}
+            case {{asHex code 4}}: // {{name}}
+            {
+                entryLength = {{size}};
+                entryOffset = static_cast<uint16_t>(entryOffset + ((index - 1) * entryLength));
+                {{#if isStruct}}
+                // Struct _{{type}}
+                _{{type}} * entry = reinterpret_cast<_{{type}} *>(write ? src : dest);
+                {{#chip_attribute_list_entryTypes}}
+                {{#if (isString type)}}
+                copyListMember(write ? dest : (uint8_t *)&entry->{{name}}, write ? (uint8_t *)&entry->{{name}} : src, write, &entryOffset, {{maxLength}}); // {{type}}
+                {{else}}
+                copyListMember(write ? dest : (uint8_t *)&entry->{{name}}, write ? (uint8_t *)&entry->{{name}} : src, write, &entryOffset, sizeof(entry->{{name}})); // {{type}}
+                {{/if}}
+                {{/chip_attribute_list_entryTypes}}
+                {{else}}
+                copyListMember(dest, src, write, &entryOffset, entryLength); // {{entryType.type}}
```

Comment:
`{{entryType.type}}` seems to be ending up as empty string in generated output.  Is that expected?